tools/generator/src/BuildSetting.swift

```diff
@@ -0,0 +1,62 @@
+/// A custom type to aid in decoding build setting dictionaries.
+///
+/// `[String: Any]` isn't `Decodable`, so we use `[String: BuildSetting]` to
+/// represent build settings in our data transfer objects. The expected use case
+/// is for a `[String: BuildSetting]` to be created with a `JSONDecoder` and for
+/// `asDictionary` to be called on it at the place of use.
+enum BuildSetting: Equatable {
+    case string(String)
+    case bool(Bool)
+    case array([String])
+}
+
+extension BuildSetting {
+    /// Converts the build setting to its native representation (e.g. `.array`
+    /// becomes an array of `String`s).
+    var asAny: Any {
+        switch self {
+        case .string(let string):
+            return string
+        case .bool(let bool):
+            return bool ? "YES" : "NO"
+        case .array(let array):
+            if array.count < 2 {
+                return array.first ?? ""
```

Comment:
ok, i'm dumb, can you explain this to me?

Reply:
If an array has 1 or 0 elements, Xcode wants it to be the same as a string.

Reply:
I'll minimally add a comment.